azurelinuxagent/ga/cgroupapi.py

```diff
@@ -389,73 +502,8 @@ def log_root_paths(self):
-
-            # We start systemd-run with shell == True so process.pid is the shell's pid, not the pid for systemd-run
-            self._systemd_run_commands.append(process.pid)
-
-        scope_name = scope + '.scope'
-
-        log_cgroup_info("Started extension in unit '{0}'".format(scope_name), send_event=False)
-
-        cpu_controller = None
-        try:
-            cgroup_relative_path = os.path.join('azure.slice/azure-vmextensions.slice', extension_slice_name)
-            cgroup = self.get_cgroup_from_relative_path(cgroup_relative_path, extension_name)
-            for controller in cgroup.get_controllers():
-                if isinstance(controller, _CpuController):
-                    cpu_controller = controller
-                CGroupsTelemetry.track_cgroup_controller(controller)
-
-        except IOError as e:
-            if e.errno == 2:  # 'No such file or directory'
-                log_cgroup_info("The extension command already completed; will not track resource usage", send_event=False)
-            log_cgroup_info("Failed to start tracking resource usage for the extension: {0}".format(ustr(e)), send_event=False)
-        except Exception as e:
-            log_cgroup_info("Failed to start tracking resource usage for the extension: {0}".format(ustr(e)), send_event=False)
-
-        # Wait for process completion or timeout
-        try:
-            return handle_process_completion(process=process, command=command, timeout=timeout, stdout=stdout,
-                                             stderr=stderr, error_code=error_code, cpu_controller=cpu_controller)
-        except ExtensionError as e:
-            # The extension didn't terminate successfully. Determine whether it was due to systemd errors or
-            # extension errors.
-            if not self._is_systemd_failure(scope, stderr):
-                # There was an extension error; it either timed out or returned a non-zero exit code. Re-raise the error
-                raise
-
-            # There was an issue with systemd-run. We need to log it and retry the extension without systemd.
-            process_output = read_output(stdout, stderr)
-            # Reset the stdout and stderr
-            stdout.truncate(0)
-            stderr.truncate(0)
-
-            if isinstance(e, ExtensionOperationError):
-                # no-member: Instance of 'ExtensionError' has no 'exit_code' member (no-member) - Disabled: e is actually an ExtensionOperationError
-                err_msg = 'Systemd process exited with code %s and output %s' % (
-                    e.exit_code, process_output)  # pylint: disable=no-member
-            else:
-                err_msg = "Systemd timed-out, output: %s" % process_output
-            raise SystemdRunError(err_msg)
-        finally:
-            with self._systemd_run_commands_lock:
-                self._systemd_run_commands.remove(process.pid)
+    def get_enabled_controllers_at_root(self):
```

Comment:
Should we add this to abstract class too? And Maybe we should add a comment in the abstract class that this only returns the agent supported controllers enabled at root (cpu and/or memory)